4-WebApp-your-API/TodoListService/Startup.cs

```diff
@@ -51,13 +51,11 @@ public void ConfigureServices(IServiceCollection services)
             // By default, the claims mapping will map claim names in the old format to accommodate older SAML applications.
             // 'http://schemas.microsoft.com/ws/2008/06/identity/claims/role' instead of 'roles'
             // This flag ensures that the ClaimsIdentity claims collection will be built from the claims in the token
-            JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
+            //JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
+
+            services.AddHttpContextAccessor();
 
             services.AddProtectedWebApi(Configuration);
-            services.Configure<JwtBearerOptions>(AzureADDefaults.JwtBearerAuthenticationScheme, options =>
```

Comment:
I thought that we want to replace this one with options.TokenValidationParameters.NameClaimType = "prefered_username" ?

Reply:
For that to work, we will need to add a manual step on Configure.ps1 to change the manifest property `"accessTokenAcceptedVersion": 2` and also have this step on the README, when the developer creates the application via portal. 
Is that fine? If it is, I would address this in another PR so all of our samples issue v2 tokens.
@kalyankrishna1, @jmprieur what do you think?

Reply:
Yes, I think that the Web API in our samples should accept v2.0 tokens.

Reply:
As discussed on our weekly meeting, we will wait for powershell to accepts changing the accepted token version to address this name claims topics. So for now we will keep as is (the step 4 sample wont mention about token versions so it wont have preferred_username) and the `User.Name` will be getting from `name` claim.